code/modules/mob/living/carbon/human/death.dm

```diff
@@ -41,9 +41,14 @@
 
 	for(var/obj/Overlays/O in obj_overlays)
 		returnToPool(O)
-	
+
 	obj_overlays = null
 
+	var/datum/species/s = species
```

Comment:
Why are you casting this?

As a result of this cast species doesn't get nulled.